src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -1048,11 +1053,16 @@ private boolean isValidAssessmentDueDate(ZonedDateTime releaseDate, ZonedDateTim
      * This method is used to validate if the previousDate is before the laterDate.
      * @return true if the previousDate is valid
      */
-    private boolean isBeforeAndNotNull(ZonedDateTime previousDate, ZonedDateTime laterDate) {
+    private static boolean isBeforeAndNotNull(ZonedDateTime previousDate, ZonedDateTime laterDate) {
         if (previousDate == null || laterDate == null) {
             return true;
         }
         return previousDate.isBefore(laterDate);
     }
 
+    private boolean isDateUsageAllowed() {
+        // We want to allow serialization to JSON for exam exercises
+        return !isExamExercise()
+                || StackWalker.getInstance().walk(stackFrames -> stackFrames.anyMatch(frame -> frame.getClassName().contains("com.fasterxml.jackson.databind.ser")));
```

Comment:
this also does not look like a good solution. What happens if the class name or serialization framework changes in the future? We would not notice the issue... I think we should avoid this in production code.

I would prefer a different solution, e.g. as part of tests or as part of a service that determines if the use of those dates is allowed